src/Reflection/ReflectionParameter.php

```diff
@@ -126,7 +126,15 @@ public static function createFromNode(
         $param->isVariadic = (bool)$node->variadic;
         $param->isByReference = (bool)$node->byRef;
         $param->parameterIndex = (int)$parameterIndex;
-        $param->typeHint = (new FindTypeFromAst())->__invoke($node->type);
+        $param->typeHint = (new FindTypeFromAst())->__invoke(
+            $node->type,
+            $function->getLocatedSource(),
+            (
+                $function instanceof ReflectionMethod
```

Comment:
yes, `ReflectionFunctionAbstract` has `getNamespaceName()` (because a function can have a namespace). Oddly enough, PHP behaviour is that if it's a `ReflectionMethod` (that inherits `ReflectionFunctionAbstract`) then the namespace will all be empty, because that information *should* be retrieved from the declaring class - hence if it's a method, you call `$function->getDeclaringClass()->getNamespaceName()` and if it's just a function you directly call `$function->getNamespaceName()`